docs/configuration/config-file-reference.md

```diff
@@ -2206,9 +2206,9 @@ The `fifo_cache_config` configures the local in-memory cache. The supported CLI
 &nbsp;
 
 ```yaml
-# The number of entries to cache.
-# CLI flag: -<prefix>.fifocache.size
-[size: <int> | default = 0]
+# Maximum memory size of the cache.
+# CLI flag: -<prefix>.fifocache.max-size-bytes
```

Comment:
Thank you for taking the time to propose this change. I'm still going over the details here, but something that has stuck out to me is that currently, this is a breaking change due to the fact that this is replacing the previous implementation based on the number of items. I feel we should be adding this size based management in addition to the count based for the time being.

Reply:
> I feel we should be adding this size based management in addition to the count based for the time being.

Should we? Technically, this (before Cortex 1.0) may be the right time to do this kind of improvements which imply having some breaking changes. This looks quite a visible breaking change to the user (Cortex will not start if the old config is passed) and doesn't look much different then the huge list of config breaking changes @gouthamve is introducing.

Reply:
I can make it not a breaking change.
For example, we can keep "size" parameter and introduce "eviction_method" string parameter, which might be either "entries" (by default) or "memory".
If set to "entries", I will follow the old logic. Otherwise new logic will be involved.

Reply:
@pracucci Even if we can because we're before 1.0, I would prefer to keep that for things we just can't get around. To me, this particular change seems like something we can phase-in but I'm open to counterpoints and if I'm the sole one thinking this way, I won't hold things up because of it.

@dmitsh I would go here with keeping both `-<prefix>.fifocache.size` and `-<prefix>.fifocache.max-size-bytes` , with the bytes flag taking precedence. This makes tracking of deprecated flag use more apparent with users.

Reply:
I updated the code per @khaines recommendations